node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala

```diff
@@ -136,8 +146,11 @@ case class NeutrinoNode(
       bestFilterHeader: CompactFilterHeaderDb,
       chainApi: ChainApi,
       bestFilterOpt: Option[CompactFilterDb]): Future[Unit] = {
+    val syncPeer = dataMessageHandler.syncPeer.getOrElse(
+      throw new RuntimeException("Sync peer not set"))
+    val syncPeerMsgSender = peerManager.peerData(syncPeer).peerMessageSender
     val sendCompactFilterHeaderMsgF = {
-      peerManager.randomPeerMsgSenderWithCompactFilters
+      syncPeerMsgSender
```

Comment:
So unless the peers are all ideal, can't really use a random one, besides without parallel sync, not much point in switching. The previous change too was by me, so fixed it now.